pkg/ruler/storage/cleaner/config.go

```diff
@@ -14,7 +15,7 @@ type Config struct {
 	Period time.Duration `yaml:"period,omitempty"`
 }
 
-func (c Config) RegisterFlags(f *flag.FlagSet) {
+func (c *Config) RegisterFlags(f *flag.FlagSet) {
```

Comment:
This pointer in config was missing, meaning that the default values for the variables were not taken into account. This might affect deployments using the ruler WAL cleaner config without any values defined.

Reply:
I think we should add an item to the upgrade guide about this

Reply:
Upgrade guide makes sense 👍 Will address this in a follow-up PR to avoid increasing the size and review of this one